contracts/Collector.sol

```diff
@@ -56,6 +56,11 @@ contract Collector is ICollector{
 
         for(uint i = 0; i < partners.length; i++)
             token.transfer(partners[i].beneficiary, SafeMath.div(SafeMath.mul(balance, partners[i].share), 100));
+        
+        balance = token.balanceOf(address(this));
```

Comment:
Why are we burning the token here? I think a lazy approach would be better because if the same partners continue to operate, the reminder may be distributed within the next `withdraw` call. In this way, we are always burning the reminder.

Reply:
completely revamped.